extensions/sns/src/main.rs

```diff
@@ -64,6 +68,7 @@ fn main() -> anyhow::Result<()> {
         SubCommand::Import(v) => commands::import::exec(v, &dfx_cache_path),
         SubCommand::Deploy(v) => commands::deploy::exec(v, &dfx_cache_path),
         SubCommand::Download(v) => commands::download::exec(v, &dfx_cache_path),
+        SubCommand::PrepareCanisters(v) => commands::prepare_canisters::exec(v, &dfx_cache_path),
```

Comment:
I forgot to tell you this, but for dfx to be able to expose a command provided by extension, you'll also need to modify `extensions/sns/extension.json` manifest, section `.subcommands`. 

`extensions.json`'s `subcommands` describes the public CLI interface of the extension; `dfx` will refuse to parse any subcommands and args that are not listed there. 
Just to avoid any misconceptions: `extension.json .subcommands` doesn't provide any glue that redirects your commands in any way; all it does is tells `dfx` _"Hey, if someone calls this extension with these subcommands and args, you are allowed to run it."_

The `about` field will be used to display help when you'd call `dfx sns --help` or `dfx sns import --help` and so on. `"long": "something-something"` tells dfx it should accept the flag `dfx <EXT> <SUBCMD> --something-something` (FYI there is also way to specify `"short": "s"` flag). Finally, you can nest subcommands like so:
```json
{
  "name": "sns",
  "...stuff...": "stuff-stuff",
  "subcommands": {
    "my-subcommand": {
      "about": "XYZ",
      "subcommands": {
         "my-nested-subcommand": {
           "about": "just like that"
         }
      } 
    }
  }
}
```

let me know if you'll need some help here

Reply:
I added and think I got it right but PTAL.

Reply:
How do I represent an argument to the extension that is not a flag arguement, e.g.:

```
dfx sns prepare-canisters add-nns-root <CANISTER_ID>
```

How do I record the `<CANISTER_ID>`?

Reply:
Looks like you found the loophole 🥲. Try running it as is, I'm curious if it fails. 

Even if it works, we would still have to change it in `dfx`, e.g. add new param to `extension.json` `.subcommands.SUBCMD.allow-values-without-hyphens<bool>`

Reply:
Does this require a new dfx release? Does this need to happen ASAP?

Reply:
BTW, it passes locally so I was able to use the `--canister-id` flag to grab the arguments in the extension and arrange them in the correct way for the underlying `sns-cli` binary. However I'm not sure diverging is the correct way to go, it would be nice to order it as closely to how the binary works as possible.